packages/data/src/components/use-select/test/index.js

```diff
@@ -132,5 +132,93 @@ describe( 'useSelect', () => {
 			children: 'bar',
 		} );
 	} );
+	describe( 'rerenders as expected with various mapSelect return ' +
+		'types', () => {
+		const getComponent = ( mapSelectSpy, dependencyKey ) => ( props ) => {
+			const dependencies = props[ dependencyKey ];
+			mapSelectSpy.mockImplementation(
```

Comment:
This implementation never needs to change. It can be a regular function.

Reply:
Do you mean `mapSelectSpy` can be a regular function?  The problem with making this change is that the test _is_ spying on the implementation here:https://github.com/WordPress/gutenberg/pull/16669/files#diff-bfc6925e9604200fafd2b38e12993a97R218

Reply:
Oh, I see, maybe it should have the same variable name to avoid confusion. They are sibling scopes so it wouldn't be shadowing.

Reply:
See f3fe2e0